File: Datasets/TrickyBugs/GenProgs/tc_generated_progs_cpp/p03042/p03042_num6_parsed.cpp
#include <iostream>
using namespace std;

int main() {
    string S;
    cin >> S;

    int a = stoi(S.substr(0, 2));
    int b = stoi(S.substr(2, 2));

    if (a > 0 && a < 13 && b > 0 && b < 13) {
        cout << "AMBIGUOUS";
    } else if ((a == 0 || a > 12) && (b == 0 || b > 12)) {
        cout << "NA";
    } else if (a == 0 || a > 12) {
        cout << "YYMM";
    } else {
        cout << "MMYY";
    }

    return 0;
}

File: Datasets/TrickyBugs/GenProgs/dpp_generated_progs_python/p03042/p03042_num6_parsed.py
S = input()

# Check if S is valid in YYMM format
if (1 <= int(S[:2]) <= 12) and (1 <= int(S[2:]) <= 99):
    yy_mm_valid = True
else:
    yy_mm_valid = False

# Check if S is valid in MMYY format
if (1 <= int(S[2:]) <= 12) and (1 <= int(S[:2]) <= 99):
    mm_yy_valid = True
else:
    mm_yy_valid = False

# Print the result based on the validity of S in different formats
if yy_mm_valid and mm_yy_valid:
    print("AMBIGUOUS")
elif yy_mm_valid:
    print("YYMM")
elif mm_yy_valid:
    print("MMYY")
else:
    print("NA")
